{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# DÃ©tection des bio-marqueurs pour le cancer colorectal\n",
    "\n",
    "## Etudiants\n",
    "- Potet Bastien\n",
    "- Kot Chau Ying"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pylab as plt\n",
    "%matplotlib inline\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from Bio import Geo\n",
    "\n",
    "def load_geo(myfile):\n",
    "    handle = open(myfile)\n",
    "    records = Geo.parse(handle)\n",
    "    return records\n",
    "\n",
    "records = load_geo('data/GSE21510_family.soft')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GEO Type: SERIES\n",
      "GEO Id: GSE21510\n",
      "Series_contact_address: 2-1-1 Hongo\n",
      "\n",
      "Series_contact_city: Bunkyo-ku\n",
      "\n",
      "Series_contact_country: Japan\n",
      "\n",
      "Series_contact_department: Intractable Disease Research Center\n",
      "\n",
      "Series_contact_email: mogushi-k@umin.ac.jp\n",
      "\n",
      "Series_contact_institute: Juntendo University\n",
      "\n",
      "Series_contact_name: Kaoru,,Mogushi\n",
      "\n",
      "Series_contact_phone: +81-3-5802-1797\n",
      "\n",
      "Series_contact_state: Tokyo\n",
      "\n",
      "Series_contact_zip/postal_code: 113-8421\n",
      "\n",
      "Series_contributor: Shunsuke,,Tsukamoto\n",
      "\n",
      "Series_contributor: Toshiaki,,Ishikawa\n",
      "\n",
      "Series_contributor: Satoru,,Iida\n",
      "\n",
      "Series_contributor: Megumi,,Ishiguro\n",
      "\n",
      "Series_contributor: Kaoru,,Mogushi\n",
      "\n",
      "Series_contributor: Hiroshi,,Mizushima\n",
      "\n",
      "Series_contributor: Hiroyuki,,Uetake\n",
      "\n",
      "Series_contributor: Hiroshi,,Tanaka\n",
      "\n",
      "Series_contributor: Kenichi,,Sugihara\n",
      "\n",
      "Series_geo_accession: GSE21510\n",
      "\n",
      "Series_last_update_date: Mar 25 2019\n",
      "\n",
      "Series_overall_design: Total of 148 microarray datasets obtaine\n",
      "d from LCM and homogenized tissues of colorectal cancer patients were normalized\n",
      " using robust multi-array average (RMA) method under R 2.6.2 statistical softwar\n",
      "e with affy package from BioConductor. Normalization was separately performed fo\n",
      "r LCM dataset and homogenized tissue dataset. The normalized gene expression lev\n",
      "els were presented as log2-transformed values by RMA.\n",
      "\n",
      "Series_platform_id: GPL570\n",
      "\n",
      "Series_platform_taxid: 9606\n",
      "\n",
      "Series_pubmed_id: 21270110\n",
      "\n",
      "Series_relation: BioProject: https://www.ncbi.nlm.nih.gov\n",
      "/bioproject/PRJNA125983\n",
      "\n",
      "Series_sample_id: GSM537330\n",
      "\n",
      "Series_sample_id: GSM537331\n",
      "\n",
      "Series_sample_id: GSM537332\n",
      "\n",
      "Series_sample_id: GSM537333\n",
      "\n",
      "Series_sample_id: GSM537334\n",
      "\n",
      "Series_sample_id: GSM537335\n",
      "\n",
      "Series_sample_id: GSM537336\n",
      "\n",
      "Series_sample_id: GSM537337\n",
      "\n",
      "Series_sample_id: GSM537338\n",
      "\n",
      "Series_sample_id: GSM537339\n",
      "\n",
      "Series_sample_id: GSM537340\n",
      "\n",
      "Series_sample_id: GSM537341\n",
      "\n",
      "Series_sample_id: GSM537342\n",
      "\n",
      "Series_sample_id: GSM537343\n",
      "\n",
      "Series_sample_id: GSM537344\n",
      "\n",
      "Series_sample_id: GSM537345\n",
      "\n",
      "Series_sample_id: GSM537346\n",
      "\n",
      "Series_sample_id: GSM537347\n",
      "\n",
      "Series_sample_id: GSM537348\n",
      "\n",
      "Series_sample_id: GSM537349\n",
      "\n",
      "Series_sample_id: GSM537350\n",
      "\n",
      "Series_sample_id: GSM537351\n",
      "\n",
      "Series_sample_id: GSM537352\n",
      "\n",
      "Series_sample_id: GSM537353\n",
      "\n",
      "Series_sample_id: GSM537354\n",
      "\n",
      "Series_sample_id: GSM537355\n",
      "\n",
      "Series_sample_id: GSM537356\n",
      "\n",
      "Series_sample_id: GSM537357\n",
      "\n",
      "Series_sample_id: GSM537358\n",
      "\n",
      "Series_sample_id: GSM537359\n",
      "\n",
      "Series_sample_id: GSM537360\n",
      "\n",
      "Series_sample_id: GSM537361\n",
      "\n",
      "Series_sample_id: GSM537362\n",
      "\n",
      "Series_sample_id: GSM537363\n",
      "\n",
      "Series_sample_id: GSM537364\n",
      "\n",
      "Series_sample_id: GSM537365\n",
      "\n",
      "Series_sample_id: GSM537366\n",
      "\n",
      "Series_sample_id: GSM537367\n",
      "\n",
      "Series_sample_id: GSM537368\n",
      "\n",
      "Series_sample_id: GSM537369\n",
      "\n",
      "Series_sample_id: GSM537370\n",
      "\n",
      "Series_sample_id: GSM537371\n",
      "\n",
      "Series_sample_id: GSM537372\n",
      "\n",
      "Series_sample_id: GSM537373\n",
      "\n",
      "Series_sample_id: GSM537374\n",
      "\n",
      "Series_sample_id: GSM537375\n",
      "\n",
      "Series_sample_id: GSM537376\n",
      "\n",
      "Series_sample_id: GSM537377\n",
      "\n",
      "Series_sample_id: GSM537378\n",
      "\n",
      "Series_sample_id: GSM537379\n",
      "\n",
      "Series_sample_id: GSM537380\n",
      "\n",
      "Series_sample_id: GSM537381\n",
      "\n",
      "Series_sample_id: GSM537382\n",
      "\n",
      "Series_sample_id: GSM537383\n",
      "\n",
      "Series_sample_id: GSM537384\n",
      "\n",
      "Series_sample_id: GSM537385\n",
      "\n",
      "Series_sample_id: GSM537386\n",
      "\n",
      "Series_sample_id: GSM537387\n",
      "\n",
      "Series_sample_id: GSM537388\n",
      "\n",
      "Series_sample_id: GSM537389\n",
      "\n",
      "Series_sample_id: GSM537390\n",
      "\n",
      "Series_sample_id: GSM537391\n",
      "\n",
      "Series_sample_id: GSM537392\n",
      "\n",
      "Series_sample_id: GSM537393\n",
      "\n",
      "Series_sample_id: GSM537394\n",
      "\n",
      "Series_sample_id: GSM537395\n",
      "\n",
      "Series_sample_id: GSM537396\n",
      "\n",
      "Series_sample_id: GSM537397\n",
      "\n",
      "Series_sample_id: GSM537398\n",
      "\n",
      "Series_sample_id: GSM537399\n",
      "\n",
      "Series_sample_id: GSM537400\n",
      "\n",
      "Series_sample_id: GSM537401\n",
      "\n",
      "Series_sample_id: GSM537402\n",
      "\n",
      "Series_sample_id: GSM537403\n",
      "\n",
      "Series_sample_id: GSM537404\n",
      "\n",
      "Series_sample_id: GSM537405\n",
      "\n",
      "Series_sample_id: GSM537406\n",
      "\n",
      "Series_sample_id: GSM537407\n",
      "\n",
      "Series_sample_id: GSM537408\n",
      "\n",
      "Series_sample_id: GSM537409\n",
      "\n",
      "Series_sample_id: GSM537410\n",
      "\n",
      "Series_sample_id: GSM537411\n",
      "\n",
      "Series_sample_id: GSM537412\n",
      "\n",
      "Series_sample_id: GSM537413\n",
      "\n",
      "Series_sample_id: GSM537414\n",
      "\n",
      "Series_sample_id: GSM537415\n",
      "\n",
      "Series_sample_id: GSM537416\n",
      "\n",
      "Series_sample_id: GSM537417\n",
      "\n",
      "Series_sample_id: GSM537418\n",
      "\n",
      "Series_sample_id: GSM537419\n",
      "\n",
      "Series_sample_id: GSM537420\n",
      "\n",
      "Series_sample_id: GSM537421\n",
      "\n",
      "Series_sample_id: GSM537422\n",
      "\n",
      "Series_sample_id: GSM537423\n",
      "\n",
      "Series_sample_id: GSM537424\n",
      "\n",
      "Series_sample_id: GSM537425\n",
      "\n",
      "Series_sample_id: GSM537426\n",
      "\n",
      "Series_sample_id: GSM537427\n",
      "\n",
      "Series_sample_id: GSM537428\n",
      "\n",
      "Series_sample_id: GSM537429\n",
      "\n",
      "Series_sample_id: GSM537430\n",
      "\n",
      "Series_sample_id: GSM537431\n",
      "\n",
      "Series_sample_id: GSM537432\n",
      "\n",
      "Series_sample_id: GSM537433\n",
      "\n",
      "Series_sample_id: GSM549099\n",
      "\n",
      "Series_sample_id: GSM549100\n",
      "\n",
      "Series_sample_id: GSM549101\n",
      "\n",
      "Series_sample_id: GSM549102\n",
      "\n",
      "Series_sample_id: GSM549103\n",
      "\n",
      "Series_sample_id: GSM549104\n",
      "\n",
      "Series_sample_id: GSM549105\n",
      "\n",
      "Series_sample_id: GSM549106\n",
      "\n",
      "Series_sample_id: GSM549107\n",
      "\n",
      "Series_sample_id: GSM549108\n",
      "\n",
      "Series_sample_id: GSM549109\n",
      "\n",
      "Series_sample_id: GSM549110\n",
      "\n",
      "Series_sample_id: GSM549111\n",
      "\n",
      "Series_sample_id: GSM549112\n",
      "\n",
      "Series_sample_id: GSM549113\n",
      "\n",
      "Series_sample_id: GSM549114\n",
      "\n",
      "Series_sample_id: GSM549115\n",
      "\n",
      "Series_sample_id: GSM549116\n",
      "\n",
      "Series_sample_id: GSM549117\n",
      "\n",
      "Series_sample_id: GSM549118\n",
      "\n",
      "Series_sample_id: GSM549119\n",
      "\n",
      "Series_sample_id: GSM549120\n",
      "\n",
      "Series_sample_id: GSM549121\n",
      "\n",
      "Series_sample_id: GSM549122\n",
      "\n",
      "Series_sample_id: GSM549123\n",
      "\n",
      "Series_sample_id: GSM549124\n",
      "\n",
      "Series_sample_id: GSM549125\n",
      "\n",
      "Series_sample_id: GSM549126\n",
      "\n",
      "Series_sample_id: GSM549127\n",
      "\n",
      "Series_sample_id: GSM549128\n",
      "\n",
      "Series_sample_id: GSM549129\n",
      "\n",
      "Series_sample_id: GSM549130\n",
      "\n",
      "Series_sample_id: GSM549131\n",
      "\n",
      "Series_sample_id: GSM549132\n",
      "\n",
      "Series_sample_id: GSM549133\n",
      "\n",
      "Series_sample_id: GSM549134\n",
      "\n",
      "Series_sample_id: GSM549135\n",
      "\n",
      "Series_sample_id: GSM549136\n",
      "\n",
      "Series_sample_id: GSM549137\n",
      "\n",
      "Series_sample_id: GSM549138\n",
      "\n",
      "Series_sample_id: GSM549139\n",
      "\n",
      "Series_sample_id: GSM549140\n",
      "\n",
      "Series_sample_id: GSM549141\n",
      "\n",
      "Series_sample_id: GSM549142\n",
      "\n",
      "Series_sample_taxid: 9606\n",
      "\n",
      "Series_status: Public on Feb 28 2011\n",
      "\n",
      "Series_submission_date: Apr 26 2010\n",
      "\n",
      "Series_summary: Purpose: This study aimed to identify a \n",
      "novel biomarker or a target of treatment for colorectal cancer (CRC).\n",
      "\n",
      "Series_summary: Experimental Design: The expression prof\n",
      "iles of cancer cells in 104 patients with CRC were examined using laser microdis\n",
      "section and oligonucleotide microarray analysis. Overexpression in CRC cells esp\n",
      "ecially in patients with distant metastasis was a prerequisite to select candida\n",
      "te genes. The mRNA expression of candidate gene was investigated by quantitative\n",
      " reversetranscription polymerase chain reaction (RT-PCR) in 77 patients as a val\n",
      "idation study. We analyzed the protein expression and localization of the candid\n",
      "ate gene by immunohistochemical study, and investigated the relationship between\n",
      " the expression and the clinicopathological feature in 274 CRC patients.\n",
      "\n",
      "Series_summary: Results: We identified 6 genes as candid\n",
      "ates related to distant metastasis in CRC patients by microarray analysis. Among\n",
      " these genes, Osteoprotegerin (OPG) is known to have an association with aggress\n",
      "iveness in several cancers through inhibiting apoptosis by neutralizing the func\n",
      "tion of tumor necrosis factor related apoptosis inducing ligand (TRAIL). The mRN\n",
      "A expression of OPG in cancer tissues was significantly higher in patients with \n",
      "distant metastasis than those without metastasis. The overexpression of OPG prot\n",
      "ein was significantly associated with worse overall survival and relapse free su\n",
      "rvival (RFS). Moreover, the overexpression of OPG protein was an independent ris\n",
      "k factor for recurrence of CRC.\n",
      "\n",
      "Series_summary: Conclusion: The overexpression of OPG wo\n",
      "uld be a predictive biomarker of recurrence and a target of the treatment for CR\n",
      "C.\n",
      "\n",
      "Series_supplementary_file: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GS\n",
      "E21nnn/GSE21510/suppl/GSE21510_RAW.tar\n",
      "\n",
      "Series_title: Clinical Significance of Osteoprotegerin\n",
      " Expression in Human Colorectal Cancer\n",
      "\n",
      "Series_type: Expression profiling by array\n",
      "\n",
      "Column Header Definitions\n",
      "\n",
      "\n",
      "---------------------- ENTITY_ATTRIBUTES:------------------------------\n",
      "\n",
      "{'Series_title': 'Clinical Significance of Osteoprotegerin Expression in Human Colorectal Cancer', 'Series_geo_accession': 'GSE21510', 'Series_status': 'Public on Feb 28 2011', 'Series_submission_date': 'Apr 26 2010', 'Series_last_update_date': 'Mar 25 2019', 'Series_pubmed_id': '21270110', 'Series_summary': ['Purpose: This study aimed to identify a novel biomarker or a target of treatment for colorectal cancer (CRC).', 'Experimental Design: The expression profiles of cancer cells in 104 patients with CRC were examined using laser microdissection and oligonucleotide microarray analysis. Overexpression in CRC cells especially in patients with distant metastasis was a prerequisite to select candidate genes. The mRNA expression of candidate gene was investigated by quantitative reversetranscription polymerase chain reaction (RT-PCR) in 77 patients as a validation study. We analyzed the protein expression and localization of the candidate gene by immunohistochemical study, and investigated the relationship between the expression and the clinicopathological feature in 274 CRC patients.', 'Results: We identified 6 genes as candidates related to distant metastasis in CRC patients by microarray analysis. Among these genes, Osteoprotegerin (OPG) is known to have an association with aggressiveness in several cancers through inhibiting apoptosis by neutralizing the function of tumor necrosis factor related apoptosis inducing ligand (TRAIL). The mRNA expression of OPG in cancer tissues was significantly higher in patients with distant metastasis than those without metastasis. The overexpression of OPG protein was significantly associated with worse overall survival and relapse free survival (RFS). Moreover, the overexpression of OPG protein was an independent risk factor for recurrence of CRC.', 'Conclusion: The overexpression of OPG would be a predictive biomarker of recurrence and a target of the treatment for CRC.'], 'Series_overall_design': 'Total of 148 microarray datasets obtained from LCM and homogenized tissues of colorectal cancer patients were normalized using robust multi-array average (RMA) method under R 2.6.2 statistical software with affy package from BioConductor. Normalization was separately performed for LCM dataset and homogenized tissue dataset. The normalized gene expression levels were presented as log2-transformed values by RMA.', 'Series_type': 'Expression profiling by array', 'Series_contributor': ['Shunsuke,,Tsukamoto', 'Toshiaki,,Ishikawa', 'Satoru,,Iida', 'Megumi,,Ishiguro', 'Kaoru,,Mogushi', 'Hiroshi,,Mizushima', 'Hiroyuki,,Uetake', 'Hiroshi,,Tanaka', 'Kenichi,,Sugihara'], 'Series_sample_id': ['GSM537330', 'GSM537331', 'GSM537332', 'GSM537333', 'GSM537334', 'GSM537335', 'GSM537336', 'GSM537337', 'GSM537338', 'GSM537339', 'GSM537340', 'GSM537341', 'GSM537342', 'GSM537343', 'GSM537344', 'GSM537345', 'GSM537346', 'GSM537347', 'GSM537348', 'GSM537349', 'GSM537350', 'GSM537351', 'GSM537352', 'GSM537353', 'GSM537354', 'GSM537355', 'GSM537356', 'GSM537357', 'GSM537358', 'GSM537359', 'GSM537360', 'GSM537361', 'GSM537362', 'GSM537363', 'GSM537364', 'GSM537365', 'GSM537366', 'GSM537367', 'GSM537368', 'GSM537369', 'GSM537370', 'GSM537371', 'GSM537372', 'GSM537373', 'GSM537374', 'GSM537375', 'GSM537376', 'GSM537377', 'GSM537378', 'GSM537379', 'GSM537380', 'GSM537381', 'GSM537382', 'GSM537383', 'GSM537384', 'GSM537385', 'GSM537386', 'GSM537387', 'GSM537388', 'GSM537389', 'GSM537390', 'GSM537391', 'GSM537392', 'GSM537393', 'GSM537394', 'GSM537395', 'GSM537396', 'GSM537397', 'GSM537398', 'GSM537399', 'GSM537400', 'GSM537401', 'GSM537402', 'GSM537403', 'GSM537404', 'GSM537405', 'GSM537406', 'GSM537407', 'GSM537408', 'GSM537409', 'GSM537410', 'GSM537411', 'GSM537412', 'GSM537413', 'GSM537414', 'GSM537415', 'GSM537416', 'GSM537417', 'GSM537418', 'GSM537419', 'GSM537420', 'GSM537421', 'GSM537422', 'GSM537423', 'GSM537424', 'GSM537425', 'GSM537426', 'GSM537427', 'GSM537428', 'GSM537429', 'GSM537430', 'GSM537431', 'GSM537432', 'GSM537433', 'GSM549099', 'GSM549100', 'GSM549101', 'GSM549102', 'GSM549103', 'GSM549104', 'GSM549105', 'GSM549106', 'GSM549107', 'GSM549108', 'GSM549109', 'GSM549110', 'GSM549111', 'GSM549112', 'GSM549113', 'GSM549114', 'GSM549115', 'GSM549116', 'GSM549117', 'GSM549118', 'GSM549119', 'GSM549120', 'GSM549121', 'GSM549122', 'GSM549123', 'GSM549124', 'GSM549125', 'GSM549126', 'GSM549127', 'GSM549128', 'GSM549129', 'GSM549130', 'GSM549131', 'GSM549132', 'GSM549133', 'GSM549134', 'GSM549135', 'GSM549136', 'GSM549137', 'GSM549138', 'GSM549139', 'GSM549140', 'GSM549141', 'GSM549142'], 'Series_contact_name': 'Kaoru,,Mogushi', 'Series_contact_email': 'mogushi-k@umin.ac.jp', 'Series_contact_phone': '+81-3-5802-1797', 'Series_contact_department': 'Intractable Disease Research Center', 'Series_contact_institute': 'Juntendo University', 'Series_contact_address': '2-1-1 Hongo', 'Series_contact_city': 'Bunkyo-ku', 'Series_contact_state': 'Tokyo', 'Series_contact_zip/postal_code': '113-8421', 'Series_contact_country': 'Japan', 'Series_supplementary_file': 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE21nnn/GSE21510/suppl/GSE21510_RAW.tar', 'Series_platform_id': 'GPL570', 'Series_platform_taxid': '9606', 'Series_sample_taxid': '9606', 'Series_relation': 'BioProject: https://www.ncbi.nlm.nih.gov/bioproject/PRJNA125983'}\n"
     ]
    }
   ],
   "source": [
    "nr = next(records)\n",
    "print(nr)\n",
    "print('\\n---------------------- ENTITY_ATTRIBUTES:------------------------------\\n')\n",
    "print(nr.entity_attributes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# start by re-loading the data\n",
    "records = load_geo('data/GSE21510_family.soft')\n",
    "\n",
    "\n",
    "id_cond = {}\n",
    "\n",
    "dict_gene = dict()\n",
    "\n",
    "data = []\n",
    "\n",
    "# loop over records\n",
    "for r in records:\n",
    "\n",
    "    rea = r.entity_attributes\n",
    "\n",
    "    if 'Sample_characteristics_ch1' in rea :\n",
    "        sample = rea['Sample_description'][1]\n",
    "        desc = rea[\"Sample_characteristics_ch1\"]\n",
    "        id_cond[sample] = desc\n",
    "\n",
    "    if 'sample_table_begin' in rea:\n",
    "        data.append(r.table_rows)\n",
    "\n",
    "    if 'platform_table_begin' in rea :\n",
    "        for i in r.table_rows :\n",
    "            dict_gene[i[0]] = i[1]\n",
    "\n",
    "\n",
    "# converts data to an array (more practical to handle)\n",
    "data = np.array(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "metastatis_list = []\n",
    "stage_list = []\n",
    "tissue_list = []\n",
    "\n",
    "for k in id_cond:\n",
    "    v = id_cond[k]\n",
    "    metastatis = v[0].split(':')[1].strip()\n",
    "    stage = v[1].split(':')[1].strip()\n",
    "    tissue = v[2].split(':')[1].split(',')[0].strip()\n",
    "    \n",
    "    metastatis_list.append(metastatis)\n",
    "    stage_list.append(stage)\n",
    "    tissue_list.append(tissue)\n",
    "    \n",
    "    id_cond[k] = [metastatis, stage, tissue]\n",
    "\n",
    "metastatis_list = list(set(metastatis_list))\n",
    "stage_list = list(set(stage_list))\n",
    "tissue_list = list(set(tissue_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['metastasis', 'none', 'metastatic recurrence']\n",
      "['3c', '3a', '1', '2b', '3b', '2a', '4', '0']\n",
      "['normal', 'cancer']\n"
     ]
    }
   ],
   "source": [
    "print(metastatis_list)\n",
    "print(stage_list)\n",
    "print(tissue_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(148, 144, 2)\n",
      "(148, 6, 2)\n"
     ]
    }
   ],
   "source": [
    "import random\n",
    "k = 6\n",
    "idx_test = random.sample(range(len(id_cond)), k)\n",
    "idx_train = list(range(0, len(id_cond) + 2))\n",
    "[idx_train.remove(i) for i in idx_test]\n",
    "\n",
    "data_test = data[:, idx_test]\n",
    "data = data[:, idx_train]\n",
    "\n",
    "print(np.shape(data))\n",
    "print(np.shape(data_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "desired_profile = ['M87338', 'X51757', 'X69699', 'L36861', 'L13852', 'X55005', 'X79510', 'J02843', 'X75208', 'L38487', 'M33318', 'NM_005505', 'NM_015140', 'NM_052871', 'NM_080735', 'NM_138957', 'NM_138957', 'NM_145004', 'NM_138796', 'NM_153359', 'NM_153359', 'BC014479', 'BG573647', 'NM_080432', 'NM_080655', 'NM_080669', 'AK074161', 'NM_138379', 'NM_173596', 'NM_024786', 'NM_080653', 'NM_001132', 'BC034926', 'BC034926', 'NM_017861', 'NM_152774', 'NM_152264', 'NM_153274', 'NM_145025', 'NM_032854', 'NM_145041', 'NM_145041', 'NM_032834', 'BC033730', 'NM_153211', 'NM_144573', 'NM_144597', 'NM_032753', 'NM_005927', 'NM_172098', 'NM_130759', 'AK091818', 'NM_144506', 'NM_033124', 'NM_033124', 'NM_138819', 'NM_138819', 'NM_145020', 'NM_145020', 'BC029352', 'NM_145039', 'NM_138632', 'BG571343', 'NM_053054', 'NM_014621', 'AY177407', 'NM_152860', 'NM_054026', 'NM_145311', 'NM_152891', 'NM_152891', 'NM_152769', 'NM_152769', 'AL833526', 'NM_052971', 'BC001526', 'NM_033128', 'NM_080618', 'NM_173487', 'BC016358', 'BC016358', 'NM_032433', 'NM_145301', 'NM_172037', 'BG677577', 'NM_080743', 'NM_153606', 'NM_153606', 'NM_152687', 'NM_152514', 'NM_173549', 'NM_173549', 'NM_152610', 'NM_152512', 'NM_152512', 'NM_052841', 'NM_080827', 'NM_138337', 'NM_145696', 'NM_152335', 'NM_138332', 'NM_138705', 'NM_153447', 'NM_153007', 'NM_153007', 'NM_173535', 'NM_152251', 'NM_152251', 'NM_147198', 'NM_147198', 'NM_152905', 'NM_153254', 'NM_153254', 'NM_145046', 'NM_145022', 'NM_005240', 'NM_152467', 'NM_152467', 'NM_078474', 'NM_145312', 'NM_170710', 'BC040487', 'NM_052836', 'NM_144708', 'NM_032445', 'BC034761', 'NM_153448', 'NM_054017', 'NM_145651', 'NM_173650', 'NM_173479', 'NM_173479', 'NM_138818', 'NM_144614', 'NM_052866', 'NM_145208', 'NM_145208', 'NM_152630', 'NM_080475', 'NM_145033', 'NM_148676', 'NM_148676']\n"
     ]
    }
   ],
   "source": [
    "desired_profile = [dict_gene[s] for s in data[0,2:,0]]\n",
    "print('desired_profile = '+str(desired_profile))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}